### PR TITLE
Removed 'compare' functions and other unnecessary bits

### PR DESCRIPTION
**Pull request by joshuahhh**
Per #9. Thanks!

### CODE REVIEW

**Comment by IjzerenHein** [comment]
Awesome, I will review and test this shortly!
Thanks for the contribution!

File: src/strength.ts [hunk 1]
```diff
@@ -13,11 +13,11 @@
     /**
      * Create a new symbolic strength.
      *
-     * @param {Number} a strong
-     * @param {Number} b medium
-     * @param {Number} c weak
-     * @param {Number} [w] weight
-     * @return {Number} strength
+     * @param a strong
```
Comment (adamhaile) [inline src/strength.ts:16]: Just curious, why remove the jsdoc type annotations?

**Comment by adamhaile** [comment]
Hey, I'm the guy who pushed #8 and made that code obsolete.  Yeah, this totally makes sense.  My initial push intentionally didn't change any code beyond maptype.ts -- I was changing enough in @IjzerenHein 's code already so preserved the code that called into `createMap()`.  Since that change has been accepted, this cleanup makes perfect sense. :+1:

The old map used an associative array internally, and the supplied `compare` function was used to sort items such that lookups were O(log N).  The new one uses a non-sorted array and an index map for O(1) lookups.  `compare` therefore isn't needed.

File: src/strength.ts (hunk 1 above)
Comment (joshuahhh) [reply src/strength.ts:16]: Reason is that the types are already present as TypeScript argument annotations. But I don't feel strongly about this at all. What happened is that my lint tool called this out as an unnecessary redundancy, and it sounded like a reasonable change to me.

(FWIW, use of jsdoc is pretty inconsistent across this codebase. I would be happy to follow whatever guidelines people want.)

**Comment by joshuahhh** [comment]
Thanks for taking a look, @adamhaile! (And for #8 too.)

**Comment by joshuahhh** [comment]
How do you think this looks, @IjzerenHein?

**Pull request merged**